File: schemesmart-ai/src/ai/agentic/form_schemas/housing_shelter/homesForIntellectuallyImpaired.py
homesForIntellectuallyImpaired = {
    "scheme_id": "HOUSING_001",
    "scheme_name": "Homes For Intellectually Impaired Persons",

    "form_steps": [

        {
            "step_id": "child_personal_information",
            "fields": [
                {"id": "child_full_name", "type": "string", "required": True},
                {"id": "date_of_birth", "type": "date", "required": True},
                {"id": "age", "type": "number", "required": True},
                {
                    "id": "gender",
                    "type": "enum",
                    "options": ["male", "female", "other"],
                    "required": True
                },
                {"id": "aadhaar_number", "type": "string", "required": False},
                {
                    "id": "orphan_status",
                    "type": "boolean",
                    "required": True
                },
                {
                    "id": "child_photo_uploaded",
                    "type": "boolean",
                    "required": True
                }
            ]
        },

        {
            "step_id": "guardian_details",
            "fields": [
                {"id": "guardian_name", "type": "string", "required": False},
                {
                    "id": "guardian_relationship",
                    "type": "enum",
                    "options": [
                        "father",
                        "mother",
                        "relative",
                        "legal_guardian",
                        "caretaker",
                        "child_welfare_committee"
                    ],
                    "required": False
                },
                {"id": "guardian_mobile_number", "type": "string", "required": False},
                {"id": "guardian_aadhaar_number", "type": "string", "required": False},
                {"id": "guardian_address", "type": "string", "required": False}
            ]
        },

        {
            "step_id": "disability_details",
            "fields": [
                {
                    "id": "disability_type",
                    "type": "enum",
                    "options": [
                        "intellectual_disability",
                        "mental_retardation",
                        "developmental_disorder"
                    ],
                    "required": True
                },
                {
                    "id": "disability_percentage",
                    "type": "number",
                    "required": True
                },
                {
                    "id": "disability_certificate_issued",
                    "type": "boolean",
                    "required": True
                },
                {
                    "id": "disability_certificate_number",
                    "type": "string",
                    "required": False
                },
                {
                    "id": "issuing_medical_authority",
                    "type": "string",
                    "required": False
                }
            ]
        },

        {
            "step_id": "residency_details",
            "fields": [
                {
                    "id": "citizenship",
                    "type": "enum",
                    "options": ["indian"],
                    "required": True
                },
                {
                    "id": "state_of_residence",
                    "type": "enum",
                    "options": ["maharashtra"],
                    "required": True
                },
                {"id": "district", "type": "string", "required": True},
                {"id": "village_or_city", "type": "string", "required": True},
                {"id": "full_address", "type": "string", "required": True},
                {
                    "id": "maharashtra_domicile_certificate",
                    "type": "boolean",
                    "required": True
                }
            ]
        },

        {
            "step_id": "child_welfare_committee_details",
            "fields": [
                {
                    "id": "referred_by_child_welfare_committee",
                    "type": "boolean",
                    "required": True
                },
                {"id": "committee_district", "type": "string", "required": False},
                {"id": "referral_order_number", "type": "string", "required": False},
                {"id": "date_of_referral", "type": "date", "required": False}
            ]
        },

        {
            "step_id": "bank_details",
            "fields": [
                {"id": "bank_account_holder_name", "type": "string", "required": True},
                {"id": "bank_account_number", "type": "string", "required": True},
                {"id": "bank_ifsc", "type": "string", "required": True},
                {"id": "bank_name", "type": "string", "required": True},
                {"id": "branch_name", "type": "string", "required": False},
                {
                    "id": "aadhaar_linked_to_bank",
                    "type": "boolean",
                    "required": False
                }
            ]
        },

        {
            "step_id": "document_upload_confirmation",
            "fields": [
                {
                    "id": "aadhaar_uploaded",
                    "type": "boolean",
                    "required": True
                },
                {
                    "id": "passport_photos_uploaded",
                    "type": "boolean",
                    "required": True
                },
                {
                    "id": "age_proof_uploaded",
                    "type": "boolean",
                    "required": True
                },
                {
                    "id": "domicile_certificate_uploaded",
                    "type": "boolean",
                    "required": True
                },
                {
                    "id": "disability_certificate_uploaded",
                    "type": "boolean",
                    "required": True
                },
                {
                    "id": "bank_passbook_uploaded",
                    "type": "boolean",
                    "required": True
                }
            ]
        },

        {
            "step_id": "declaration",
            "fields": [
                {
                    "id": "information_true",
                    "type": "boolean",
                    "required": True
                },
                {
                    "id": "guardian_signature_name",
                    "type": "string",
                    "required": True
                },
                {
                    "id": "submission_date",
                    "type": "date",
                    "required": True
                }
            ]
        }

    ]
}

File: schemesmart-ai/src/ai_agentic/form_schemas/housing/homesForIntellectuallyImpaired.ts
export const homesForIntellectuallyImpaired = {
  scheme_id: "HOUSING_001",

  scheme_name: "Homes For Intellectually Impaired Persons",

  eligibility_rules: {
    min_disability_percentage: 40,
    required_disability_types: ["mentally_deficient_child"],
    max_age: null,
    min_age: null,
    allowed_categories: null,
    income_field: null,
    category_field: null,
    age_field: "age",
    disability_field: "disability_percentage",
    disability_type_field: "disability_type",
  },

  form_steps: [
    {
      step_id: "applicant_details",
      step_label: "Applicant (Child) Details",

      fields: [
        {
          id: "child_full_name",
          label: "Child's Full Name",
          placeholder: "Enter the child's full name",
          hint: "Full name of the intellectually impaired child as per birth certificate or Aadhaar.",
          type: "string",
          required: true,
        },

        {
          id: "date_of_birth",
          label: "Date of Birth",
          placeholder: "",
          hint: "Child's date of birth as per official records.",
          type: "date",
          required: true,
        },

        {
          id: "age",
          label: "Age (in years)",
          placeholder: "e.g. 12",
          hint: "Current age of the child in completed years.",
          type: "number",
          required: true,
          eligibility_key: "age",
        },

        {
          id: "gender",
          label: "Gender",
          placeholder: "",
          hint: "Gender of the child as per official documents.",
          type: "enum",
          options: ["male", "female", "other"],
          required: true,
        },
      ],
    },

    {
      step_id: "disability_details",
      step_label: "Disability Details",

      fields: [
        {
          id: "disability_type",
          label: "Type of Disability",
          placeholder: "",
          hint: "This scheme is specifically for mentally deficient (intellectually impaired) children.",
          type: "enum",
          options: ["mentally_deficient_child"],
          required: true,
          eligibility_key: "required_disability_types",
        },

        {
          id: "disability_percentage",
          label: "Disability Percentage (%)",
          placeholder: "e.g. 50",
          hint: "As certified by the medical authority. Must be at least 40% to qualify for this scheme.",
          type: "number",
          required: true,
          eligibility_key: "min_disability_percentage",
        },

        {
          id: "disability_certificate_available",
          label: "Disability Certificate Available",
          placeholder: "",
          hint: "A valid disability certificate issued by a government-recognized medical authority is mandatory.",
          type: "boolean",
          required: true,
        },
      ],
    },

    {
      step_id: "guardian_details",
      step_label: "Guardian / Parent Details",

      fields: [
        {
          id: "guardian_name",
          label: "Guardian's Full Name",
          placeholder: "Enter guardian's name",
          hint: "Full name of the parent or legal guardian of the child.",
          type: "string",
          required: true,
        },

        {
          id: "guardian_mobile",
          label: "Guardian's Mobile Number",
          placeholder: "10-digit mobile number",
          hint: "Active mobile number of the guardian for communication purposes.",
          type: "string",
          required: true,
        },

        {
          id: "guardian_relationship",
          label: "Relationship with Child",
          placeholder: "e.g. Father, Mother, Legal Guardian",
          hint: "Specify the guardian's relationship with the child.",
          type: "string",
          required: true,
        },
      ],
    },

    {
      step_id: "bank_details",
      step_label: "Bank Details",

      fields: [
        {
          id: "bank_account_number",
          label: "Bank Account Number",
          placeholder: "Enter your account number (9–18 digits)",
          hint: "Bank account number of the guardian where benefits will be transferred.",
          type: "string",
          required: true,
        },

        {
          id: "bank_ifsc",
          label: "Bank IFSC Code",
          placeholder: "e.g. SBIN0001234",
          hint: "11-character IFSC code found on your cheque book or passbook.",
          type: "string",
          required: true,
        },
      ],
    },
  ],
};